fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RelativePriorityPlanSelector.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * RelativePriority.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.plans;
+
+import com.apple.foundationdb.record.ObjectPlanHash;
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.RecordCoreArgumentException;
+import com.google.common.annotations.VisibleForTesting;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * A utility class to implement an algorithm to make a random selection based off of relative priorities.
+ * The priorities are given in the list of priorities (to the constructor) and the selector chooses a number (between
+ * 0 and the priorities.size - 1), where the chance for a given selection to be picked being relative to its value.
+ * The relative priorities are given as % points: they should all add up to 100.
```

Comment:
Why not use fractions (doubles), as an earlier comment suggests?

Reply:
That was the original implementation, but due to PR comments (the main concern was with rounding issues with doubles), this seemed like a more acceptable solution.
The alternative would have been an unbound list of numbers (either fixed or floating point) whose sum would have been the "100%". The issue I had with this approach is that it would have been harder for a user to figure out the exact relative probabilities (esp. after a while, where the numbers may fluctuate and the total become arbitrary).